### PR TITLE
Various refactorings

### PR DESCRIPTION
**Pull request by josephcsible**

### CODE REVIEW

File: src/ShellCheck/Analytics.hs [hunk 1]
```diff
@@ -3858,13 +3858,13 @@ groupByLink :: (a -> a -> Bool) -> [a] -> [[a]]
 groupByLink f list =
     case list of
         [] -> []
-        (x:xs) -> g x [] xs
+        (x:xs) -> foldr c n xs x []
   where
-    g current span (next:rest) =
+    c next rest current span =
```
Comment (Tylerwbrown) [inline src/ShellCheck/Analytics.hs:3863]: Why was this changed from `g` to `c`?

**Review by Tylerwbrown — APPROVED** [review]
Pretty much all these changes are an improvement in my opinion, for what it's worth.

File: src/ShellCheck/Analytics.hs (hunk 1 above)
Comment (josephcsible) [reply src/ShellCheck/Analytics.hs:3863]: Before, `g` handled both the cons and nil cases, since it was doing explicit recursion. Now that `foldr` is handling the recursion, there's two different functions, so I called them `c` for the cons case and `n` for the nil case.

**Comment by koalaman** [comment]
Thanks!

File: shellcheck.hs [hunk 2]
```diff
@@ -507,7 +507,7 @@ ioInterface options files = do
       where
         find filename deflt = do
             sources <- findM ((allowable inputs) `andM` doesFileExist) $
-                        (adjustPath filename):(map (</> filename) $ map adjustPath $ sourcePathFlag ++ sourcePathAnnotation)
```
Comment (demkaid) [inline shellcheck.hs:510]: - [ ] 

- [ ] `[](url)`
